plugin.picture.googlephotos/main.py

```diff
@@ -0,0 +1,395 @@
+            xbmc.log(str(status_code), xbmc.LOGDEBUG)
+        time = int((1 - (datetime.datetime.utcnow() -
+                         init_time).total_seconds() / code_json["expires_in"]) * 100)
+        login_dialog.update(time)
+
+        if time <= min_time_to_refresh or status_code == 400:
+            code_json = get_device_code()
+            min_time_to_refresh = (
+                100 / code_json["expires_in"]) * code_json['interval']
+            init_time = datetime.datetime.utcnow()
+            sleep_time = code_json['interval'] * 1000
+            login_dialog.update(time, message=dialog_msg +
+                                code_json["userCode"])
+            xbmc.executebuiltin(
+                f'Notification({__addon__.getLocalizedString(30403)}, ,time=3000)')
+            time = 100
+    # except Exception as exec:
+    # xbmc.log(str(exec), xbmc.LOGDEBUG)
+    # err_dialog = xbmcgui.Dialog()
+    # err_dialog.notification(__addon__.getLocalizedString(30411),
+    #                         __addon__.getLocalizedString(30422),
+    #                         xbmcgui.NOTIFICATION_ERROR, 3000)
+    # finally:
+    login_dialog.close()
+    xbmc.executebuiltin('Container.Refresh')
+
+
+def remove_account():
+    xbmcvfs.delete(str(token_path))
+    xbmc.executebuiltin('Container.Refresh')
+
+
+def list_options():
+    items = []
+
+    # Third string in the tuples are API endpoint's route
+    modes = [(__addon__.getLocalizedString(30404), 'list_media', 'all'),
+             (__addon__.getLocalizedString(30405), 'list_albums', 'albums'),
+             (__addon__.getLocalizedString(30406), 'list_albums', 'sharedAlbums'),
+             (__addon__.getLocalizedString(30407), 'custom_filter')]
+
+    for mode in modes:
+        if len(mode) == 2:
+            url = utils.build_url(
+                base_url, {'mode': mode[1], 'token_filename': token_path.name})
+        else:
+            url = utils.build_url(
+                base_url, {'mode': mode[1], 'type': mode[2], 'token_filename': token_path.name})
+        li = xbmcgui.ListItem(mode[0])
+        items.append((url, li, True))  # (url, listitem[, isFolder])
+
+    xbmcplugin.addDirectoryItems(addon_handle, items)
+    xbmcplugin.endOfDirectory(addon_handle)
+
+
+def refresh():
+    '''
+    Refreshes the current window
+    IMP: Pass previous query string as argument in the url with key prev_q
+    '''
+    id = xbmcgui.getCurrentWindowId()
+    pos = int(xbmc.getInfoLabel(f'Container.CurrentItem'))
+    cont_path = utils.build_url(
+        base_url, {'call_type': 1}, args.get('prev_q')[0])
+    xbmc.executebuiltin(f'Container.Update({cont_path})')
+
+    # get active window
+    win = xbmcgui.Window(id)
+    cid = win.getFocusId()
+    # Check if window is fully loaded
+    while not xbmc.getCondVisibility(f'Control.IsVisible({cid})'):
+        xbmc.sleep(150)
+    # Focus on the last focused position
+    xbmc.executebuiltin(f'SetFocus({cid},{pos},absolute)')
```

Comment:
I guess that `xbmcgui.Window().setFocusId()` does the same thing. See https://romanvm.github.io/Kodistubs/_autosummary/xbmcgui.html

Reply:
setFocusId doesn't scroll to a specific position in the list. So, I haven't made any changes here.